test/WebJobs.Script.Tests.Integration/SamplesEndToEndRestartTests.cs

```diff
@@ -55,10 +46,10 @@ public class SamplesEndToEndRestartTests
                     }));
                 }
 
+                // force some host restarts while the above requests are executing
+                // in parallel
                 string functionJsonPath = Path.Combine(fixture.HostSettings.ScriptPath, "HttpTrigger-CSharp", "function.json");
-
-                // try for 2 minutes to force a host restart
```

Comment:
Yes, then with my fix it was passing consistently. However, due to the nature of the test I could never get it to run completely reliably. I don't think a 1-2 minute test like this is something we want to have anyways. So I've removed the test completely, and instead added a unit test verifying my changes.

Feel free to pull these changes and try them out against your scenario.